### PR TITLE
AE advantage/limits, wording, typos

### PR DESCRIPTION
**Pull request by fxguenther**
Addresses issues #12, #21, #22

### CODE REVIEW

File: draft-irtf-cfrg-aead-limits.md [hunk 1]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
+
+~~~
+CA <= AEA
+IA <= AEA
+AEA <= CA + IA
+~~~
+
+Each application requires an individual determination of limits in order to keep CA
 and IA sufficiently small.  For instance, TLS aims to keep CA below 2^-60 and IA
 below 2^-57. See {{?TLS=RFC8446}}, Section 5.5.
```
Comment (martinthomson) [inline draft-irtf-cfrg-aead-limits.md:198]: ...in the single-user setting.

File: draft-irtf-cfrg-aead-limits.md [hunk 2]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
 
-Once an upper bound on CA and IA are determined, this document
-defines a process for determining two overall limits:
+Once upper bounds on CA, IA, or AEA are determined, this document
+defines a process for determining three overall operational limits:
+
+- Confidentiality limit (CL): The number of messages (possibly depending on the
+  number of bytes of plaintext and maybe authenticated additional data (AAD))
+  an application can encrypt before giving the adversary a confidentiality
+  advantage higher than CA.
```
Comment (martinthomson) [inline draft-irtf-cfrg-aead-limits.md:208]: Maybe reword:

```suggestion
- Confidentiality limit (CL): The number of messages an application can encrypt
  before giving the adversary a confidentiality advantage higher than CA.
```

Then below:

```
When limits are expressed as a number of messages, this requires assumptions about
the size of messages and any authenticated additional data (AAD).  Limits can instead
be expressed in terms of the number of plaintext bytes and maybe AAD bytes in total.
```

File: draft-irtf-cfrg-aead-limits.md [hunk 3]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
+- Integrity limit (IL): The number ciphertexts (possibly depending on the
+  number of bytes of ciphertext and maybe authenticated additional data (AAD))
+  an application can decrypt, either successfully or not,
+  before giving the adversary an integrity advantage higher than IA.
 
-- Integrity limit (IL): The number of bytes of ciphertext and maybe authenticated
-  additional data (AAD) an application can process, either successfully or not,
-  before giving the adversary a non-negligible IA.
+- Authenticated encryption limit (AEL): The combined number of messages and
+  number of ciphertext an appliation can en-/decrypt before giving the adversary
```
Comment (martinthomson) [inline draft-irtf-cfrg-aead-limits.md:216]:
```suggestion
  number of ciphertexts an application can encrypt or decrypt before giving the adversary
```

File: draft-irtf-cfrg-aead-limits.md [hunk 4]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
+  an authenticated enryption advantage higher than AEA.
```
Comment (martinthomson) [inline draft-irtf-cfrg-aead-limits.md:217]:
```suggestion
  an authenticated encryption advantage higher than AEA.
```

**Review by martinthomson — COMMENTED** [review]
This is good stuff.  Thanks for doing it.

File: draft-irtf-cfrg-aead-limits.md [hunk 5]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
 | u | Number of users or keys (multi-user setting only) |
 
-For each AEAD algorithm, we define the confidentiality and integrity advantage
-roughly as the advantage an attacker has in breaking the corresponding security
-property for the algorithm. Specifically:
+For each AEAD algorithm, we define the (passive) confidentiality and (active)
+integrity advantage roughly as the advantage an attacker has in breaking the
+corresponding classical security property for the algorithm. Moreover, we 
+define the combined authenticated encryption advantage guaranteeing both
+confidentiality and integrity against an active network attacker. Specifically:
```
Comment (chris-wood) [inline draft-irtf-cfrg-aead-limits.md:168]:
```suggestion
confidentiality and integrity against an active attacker. Specifically:
```

File: draft-irtf-cfrg-aead-limits.md [hunk 6]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
 
-- Confidentiality advantage (CA): The advantage of an attacker succeeding in breaking
-the confidentiality properties of the AEAD. In this document, the definition of
-confidentiality advantage is the increase in the probability that an attacker is
-able to successfully distinguish an AEAD ciphertext from the output of a random
-function.
+- Confidentiality advantage (CA): The probability of a network-passive attacker
```
Comment (chris-wood) [inline draft-irtf-cfrg-aead-limits.md:170]:
```suggestion
- Confidentiality advantage (CA): The probability of a passive attacker
```

File: draft-irtf-cfrg-aead-limits.md [hunk 7]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
+succeeding in breaking the confidentiality properties (IND-CPA) of the AEAD scheme.
+In this document, the definition of confidentiality advantage roughly is the
+probability that an attacker successfully distinguishes the ciphertext outputs
+of the AEAD scheme from the outputs of a random function.
 
-- Integrity advantage (IA): The probability of an attacker succeeding in breaking
-the integrity properties of the AEAD. In this document, the definition of
-integrity advantage is the probability that an attacker is able to forge a
-ciphertext that will be accepted as valid.
+- Integrity advantage (IA): The probability of a network-active attacker succeeding
```
Comment (chris-wood) [inline draft-irtf-cfrg-aead-limits.md:176]:
```suggestion
- Integrity advantage (IA): The probability of a active attacker succeeding
```

File: draft-irtf-cfrg-aead-limits.md [hunk 8]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
+in breaking the integrity properties (INT-CTXT) of the AEAD scheme. In this document,
+the definition of integrity advantage roughly is the probability that an attacker
+is able to forge a ciphertext that will be accepted as valid.
 
-Each application requires a different application of limits in order to keep CA
+- Authenticated Encryption advantage (AEA): The probability of a network-active
```
Comment (chris-wood) [inline draft-irtf-cfrg-aead-limits.md:181]:
```suggestion
- Authenticated Encryption advantage (AEA): The probability of a active
```

File: draft-irtf-cfrg-aead-limits.md [hunk 9]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
+attacker succeeding in breaking the authenticated-encryption properties of the
+AEAD scheme. In this document, the definition of authenticated encryption
+advantage roughly is the probability that an attacker successfully distinguishes
+the ciphertext outputs of the AEAD scheme from the outputs of a random function
+OR is able to forge a ciphertext that will be accepted as valid.
```
Comment (chris-wood) [inline draft-irtf-cfrg-aead-limits.md:186]:
```suggestion
or is able to forge a ciphertext that will be accepted as valid.
```

File: draft-irtf-cfrg-aead-limits.md (hunk 2 above)
Comment (chris-wood) [reply draft-irtf-cfrg-aead-limits.md:208]: +1 to this suggestion!

File: draft-irtf-cfrg-aead-limits.md [hunk 10]
```diff
@@ -155,43 +155,66 @@ This document defines limitations in part using the quantities below.
...
+  an application can encrypt before giving the adversary a confidentiality
+  advantage higher than CA.
 
-- Confidentiality limit (CL): The number of bytes of plaintext and maybe
-  authenticated additional data (AAD) an application can encrypt before giving
-  the adversary a non-negligible CA.
+- Integrity limit (IL): The number ciphertexts (possibly depending on the
+  number of bytes of ciphertext and maybe authenticated additional data (AAD))
+  an application can decrypt, either successfully or not,
+  before giving the adversary an integrity advantage higher than IA.
```
Comment (chris-wood) [inline draft-irtf-cfrg-aead-limits.md:213]:
```suggestion
- Integrity limit (IL): The number of ciphertexts an application can decrypt, 
  either successfully or not, before giving the adversary an integrity advantage 
  higher than IA.
```

**Review by chris-wood — APPROVED** [review]
This LGTM with the suggested changes! Thanks, @fxguenther. Per #21, should we also include a reference for the IND-CTXT and IND-CPA properties?

File: draft-irtf-cfrg-aead-limits.md (hunk 1 above)
Comment (fxguenther) [reply draft-irtf-cfrg-aead-limits.md]: Resolved in b7ddd01

File: draft-irtf-cfrg-aead-limits.md (hunk 2 above)
Comment (fxguenther) [reply draft-irtf-cfrg-aead-limits.md]: Resolved in b7ddd01

File: draft-irtf-cfrg-aead-limits.md (hunk 10 above)
Comment (fxguenther) [reply draft-irtf-cfrg-aead-limits.md]: Resolved in b7ddd01

**Comment by fxguenther** [comment]
> This LGTM with the suggested changes! Thanks, @fxguenther. Per #21, should we also include a reference for the IND-CTXT and IND-CPA properties?

Good point, I added references now (b7ddd01)
